tests/pubsub/testpreamblestore.nim

```diff
@@ -0,0 +1,181 @@
+import unittest2
+
+{.used.}
+
+import chronos
+import
+  ../../libp2p/[
+    peerid,
+    protocols/pubsub/gossipsub/preamblestore,
+    protocols/pubsub/gossipsub/types,
+    protocols/pubsub/rpc/message,
+    protocols/pubsub/pubsubpeer,
+    protocols/pubsub/rpc/messages,
+  ]
+import ../utils/async_tests
+import ./utils
+import ../helpers
+
+proc mockPreamble(
+    id: MessageId,
+    sender: PubSubPeer,
+    startOffset: Duration = 0.seconds,
+    expireOffset: Duration = 10.seconds,
+): PreambleInfo =
+  let now = Moment.now()
+  PreambleInfo.init(
+    ControlPreamble(messageID: id, messageLength: 123, topicID: "test"),
+    sender = sender,
+    starts = now + startOffset,
+    expiresAt = now + expireOffset,
+  )
+
+suite "preamble store":
+  teardown:
+    checkTrackers()
+
+  const topic = "foobar"
+
+  asyncTest "insert and retrieve":
+    var (gossipSub, conns, peers) = setupGossipSubWithPeers(1, topic)
+    defer:
+      await teardownGossipSub(gossipSub, conns)
```

Comment:
why is `setupGossipSubWithPeers` needed here?

Reply:
I guess it could be possible to setup a single PubSubPeer somehow?  will try that

Reply:
fixed in last commit